solend-sdk/src/state/reserveV2.ts

```diff
@@ -0,0 +1,252 @@
+import { AccountInfo, PublicKey } from "@solana/web3.js";
+import BN from "bn.js";
+import { Buffer } from "buffer";
+import * as Layout from "../utils/layout";
+import { LastUpdate, LastUpdateLayout } from "./lastUpdate";
+
+const BufferLayout = require("buffer-layout");
+
+export interface Reserve {
+  version: number;
+  lastUpdate: LastUpdate;
+  lendingMarket: PublicKey;
+  liquidity: ReserveLiquidity;
+  collateral: ReserveCollateral;
+  config: ReserveConfig;
+  rateLimiter: RateLimiter;
+}
+
+export interface ReserveLiquidity {
+  mintPubkey: PublicKey;
+  mintDecimals: number;
+  supplyPubkey: PublicKey;
+  // @FIXME: oracle option
+  oracleOption: number;
+  pythOracle: PublicKey;
+  switchboardOracle: PublicKey;
+  availableAmount: BN;
+  borrowedAmountWads: BN;
+  cumulativeBorrowRateWads: BN;
+  accumulatedProtocolFeesWads: BN;
+  marketPrice: BN;
+  smoothedMarketPrice: BN;
+}
+
+export interface ReserveCollateral {
+  mintPubkey: PublicKey;
+  mintTotalSupply: BN;
+  supplyPubkey: PublicKey;
+}
+
+export interface ReserveConfig {
+  optimalUtilizationRate: number;
+  loanToValueRatio: number;
+  liquidationBonus: number;
+  liquidationThreshold: number;
+  minBorrowRate: number;
+  optimalBorrowRate: number;
+  maxBorrowRate: number;
+  fees: {
+    borrowFeeWad: BN;
+    flashLoanFeeWad: BN;
+    hostFeePercentage: number;
+  };
+  depositLimit: BN;
+  borrowLimit: BN;
+  feeReceiver: PublicKey;
+  protocolLiquidationFee: number;
+  protocolTakeRate: number;
+  addedBorrowWeightBPS: number;
+  borrowWeight: number;
```

Comment:
why do we have this? Its not in state/reserve.rs

Reply:
+1

Reply:
borrowWeight? just because we don't use addedBorrowWeightBPS in the code, and call borrow_weight() instead.
I figured i'd just add the pre calculated value as a field for convenience